### PR TITLE
Webconfig Aker Blob Support

### PR DESCRIPTION
**Pull request by sadhyama**

### CODE REVIEW

File: src/webcfg_client.c [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+		}
+		else if(strncmp(paramName, AKER_DELETE_PARAM, sizeof(AKER_DELETE_PARAM)) ==0)
+		{
+			msg->msg_type = WRP_MSG_TYPE__DELETE;
+			msg->u.crud.payload = NULL;
+			msg->u.crud.payload_size = 0;
+		}
+		else
+		{
+			WebcfgError("Invalid aker request\n");
```
Comment (shilpa24balaji) [inline src/webcfg_client.c:121]: Lets add comment that retrieve is not supported through webconfig

File: src/webcfg_client.c [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+	}
+	else
+	{
+		WebcfgError("Failed to decode msgpack data\n");
+	}
+	msgpack_unpacked_destroy(&result);
+	return decodedPayload;
+}
+
+static char* parsePayloadForAkerStatus(char *payload)
```
Comment (shilpa24balaji) [inline src/webcfg_client.c:507]: Aker code lets make a separate file

File: src/webcfg_client.c [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+				WebcfgDebug("sourceApplication: %s\n",sourceApplication);
+				if(sourceService != NULL && sourceApplication != NULL && strcmp(sourceService,"aker")== 0 && strcmp(sourceApplication,"schedule")== 0)
+				{
+					WebcfgInfo("Response received from %s\n",sourceService);
+					WEBCFG_FREE(sourceService);
+					WEBCFG_FREE(sourceApplication);
+					payload = decodePayload(wrp_msg->u.crud.payload);
+					WebcfgDebug("payload = %s\n",payload);
+					WebcfgDebug("status: %d\n",wrp_msg->u.crud.status);
+					handleAkerStatus(wrp_msg->u.crud.status, payload);
```
Comment (shilpa24balaji) [inline src/webcfg_client.c:346]: Are we setting the aker status in the tmp error details or notification?

Comment (sadhyama) [reply src/webcfg_client.c:346]: yes this is already handled as part of NACK events in webcfg_event.c , updated error_details in tmp and send notifications.

File: src/webcfg_client.c (hunk 2 above)
Comment (sadhyama) [reply src/webcfg_client.c:507]: webcfg_client.c is a new file which has aker code to handle client send/receive based on wrp msgTypes.
If we need to rename webcfg_client.c to any other better name (probably specific to aker), that can be done.

File: src/webcfg_client.c (hunk 1 above)
Comment (sadhyama) [reply src/webcfg_client.c:121]: Incorporated.

File: src/webcfg_client.c (hunk 2 above)
Comment (shilpa24balaji) [reply src/webcfg_client.c:507]: I felt webcfg_client.c for nanomsg client is fine but for aker we can have new file

Comment (sadhyama) [reply src/webcfg_client.c]: Incorporated. Added webcfg_aker.c for aker specific code and webcfg_client.c for nanomsg client.

File: src/webcfg_aker.c [hunk 4]
```diff
@@ -0,0 +1,566 @@
...
+			        }
+				WebcfgInfo("Request:> param[%d].name = %s, type = %d\n",i,reqParam[i].name,reqParam[i].type);
+				WebcfgDebug("Request:> param[%d].value = %s\n",i,reqParam[i].value);
+			}
+
+			if(reqParam !=NULL && validate_request_param(reqParam, paramCount) == WEBCFG_SUCCESS)
+			{
+				//check aker ready and is registered to parodus using RETRIEVE request to parodus.
+				WebcfgDebug("AkerStatus to check service ready\n");
+				if(checkAkerStatus() != WEBCFG_SUCCESS)
```
Comment (shilpa24balaji) [inline src/webcfg_aker.c:308]: I feel this checkAkerStatus() should be first thing we check for. Are we sending any NACK if blob invalid before this aker status check is done?

Comment (sadhyama) [reply src/webcfg_aker.c]: moved checkAkerStatus() as first check. No, we are not sending any NACK before that, should we send anything like that?

Comment (shilpa24balaji) [reply src/webcfg_aker.c]: Not necessary to send NACK was just thinking

File: src/webcfg_aker.c [hunk 5]
```diff
@@ -0,0 +1,563 @@
...
+		if(sendStatus == 0)
+		{
+			WebcfgInfo("Sent aker retrieve request to parodus\n");
+		}
+		else
+		{
+			WebcfgError("Failed to send aker retrieve req: '%s'\n",libparodus_strerror(sendStatus));
+		}
+
+		//waiting to get response from parodus. add lock here while reading
```
Comment (shilpa24balaji) [inline src/webcfg_aker.c:193]: This block should be inside the if(sendStatus == 0)

File: src/webcfg_aker.c [hunk 6]
```diff
@@ -0,0 +1,563 @@
...
+		msg->u.crud.dest = strdup(dest);
+		transaction_uuid = generate_trans_uuid();
+		if(transaction_uuid !=NULL)
+		{
+			msg->u.crud.transaction_uuid = transaction_uuid;
+			WebcfgInfo("transaction_uuid generated is %s\n", msg->u.crud.transaction_uuid);
+		}
+		msg->u.crud.content_type = strdup(CONTENT_TYPE_JSON);
+
+		sendStatus = libparodus_send(get_webcfg_instance(), msg);
```
Comment (shilpa24balaji) [inline src/webcfg_aker.c:183]: Is there any case where parodus might not be available over nanomsg? I think wan-down or other cloud offline does not matter for aker. Can we test it once?

File: src/webcfg_aker.c [hunk 7]
```diff
@@ -0,0 +1,563 @@
...
+						WEBCFG_FREE(reqParam);
+						webcfgparam_destroy( pm );
+						return rv;
+					}
+			        }
+				WebcfgInfo("Request:> param[%d].name = %s, type = %d\n",i,reqParam[i].name,reqParam[i].type);
+				WebcfgDebug("Request:> param[%d].value = %s\n",i,reqParam[i].value);
+			}
+
+			if(reqParam !=NULL && validate_request_param(reqParam, paramCount) == WEBCFG_SUCCESS)
```
Comment (shilpa24balaji) [inline src/webcfg_aker.c:313]: Seems like if validate_request_param fails we are not sending any failure notification.

Comment (sadhyama) [reply src/webcfg_aker.c:304]: yes, here we are not sending notification for any docs yet.

File: src/webcfg_aker.c (hunk 5 above)
Comment (sadhyama) [reply src/webcfg_aker.c]: done.

File: src/webcfg_aker.c (hunk 6 above)
Comment (sadhyama) [reply src/webcfg_aker.c:183]: Tested wan down case and confirmed aker doc send receive are success over nanosmg.

**Pull request merged**